src/support/auth/authentication-manager.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { isObject } from '@adobe/spacecat-shared-utils';
+
+import NotAuthenticatedError from './errors/not-authenticated.js';
+
+/**
+ * Authentication manager. It will try to authenticate the request with all the provided handlers.
+ * If none of the handlers are able to authenticate the request, it will throw
+ * a NotAuthenticatedError.
+ * @class
+ */
+export default class AuthenticationManager {
+  constructor(log) {
+    this.log = log;
+    this.handlers = [];
+  }
+
+  /**
+   * Register a handler. This method is private and should not be called directly.
+   * The handlers are used in the order they are registered.
+   * @param {AbstractHandler} Handler - The handler to be registered
+   */
+  #registerHandler(Handler) {
+    this.handlers.push(new Handler(this.log));
+  }
+
+  /**
+   * Authenticate the request with all the handlers.
+   * @param {Object} request - The request object
+   * @param {UniversalContext} context - The context object
+   * @return {Promise<AuthInfo>} The authentication info
+   * @throws {NotAuthenticatedError} If no handler was able to authenticate the request
+   */
+  async authenticate(request, context) {
+    for (const handler of this.handlers) {
```

Comment:
looping through handlers doesn't seem to be the best way to all these. Especially when `handler.checkAuth` needs to perform some http call for an auth service like IMS.

wouldn't it be possible to detect what authenticating mechanism user wants to use? Ie. we can look at the `x-api-key` header. If it exists, then user wants to authenticate via legacy.

helix-admin does a similar thing: https://github.com/adobe/helix-admin/blob/main/src/support/auth.js#L560